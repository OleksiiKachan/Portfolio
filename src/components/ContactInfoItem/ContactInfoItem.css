.contactInfo__contactItem {
    margin: 30px 0;
	display: flex;
	align-items: center;
}

.contactInfo__icon {
    margin-right: 25px;
}