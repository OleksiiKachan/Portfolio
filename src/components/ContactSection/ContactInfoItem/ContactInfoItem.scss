.contactInfoItem {
  margin: 30px 0;
  display: flex;
  align-items: center;

  &__icon {
    margin-right: 25px;
  }
}
