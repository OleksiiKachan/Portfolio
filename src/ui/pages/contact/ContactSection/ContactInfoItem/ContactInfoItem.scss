@import '../../../../assets/style/variables';

.contactInfoItem {
  margin: 30px 0;
  display: flex;
  align-items: center;
  text-decoration: none;
  color: $color-black;
  font-weight: 400;
  font-style: normal;

  &:hover {
    text-decoration: underline;
  }

  &__icon {
    margin-right: 25px;
  }
}
