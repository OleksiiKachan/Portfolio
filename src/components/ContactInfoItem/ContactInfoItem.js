import React, { Component } from 'react';
import classNames from 'classnames';
import './ContactInfoItem.scss';

/*
    <ContactInfoItem icon='' text='' className />
*/
class ContactInfoItem extends Component {
    render() {
        return (
            <div className={classNames('contactInfo__contactItem', this.props.className)}>
                <img className={classNames('contactInfo__icon')} src={this.props.icon} alt='phone' />
                <p className={classNames('contactInfo__text')}>{this.props.text}</p>
            </div>
        );
    }
}

export default ContactInfoItem;