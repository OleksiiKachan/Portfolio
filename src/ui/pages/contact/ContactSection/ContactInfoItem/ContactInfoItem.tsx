import React from 'react';
import classNames from 'classnames';
import './ContactInfoItem.scss';

/*
    <ContactInfoItem
      icon=''
      content=''
      className=''
    />
*/

type PropsType = {
  icon: string;
  iconAlt: string;
  content: string;
  className?: string;
};

export default ({ icon, iconAlt, content, className }: PropsType) => {
  return (
    <div className={classNames('contactInfoItem', className)}>
      <img
        className={classNames('contactInfoItem__icon')}
        src={icon}
        alt={iconAlt}
      />
      <p className={classNames('contactInfoItem__content')}>{content}</p>
    </div>
  );
};
