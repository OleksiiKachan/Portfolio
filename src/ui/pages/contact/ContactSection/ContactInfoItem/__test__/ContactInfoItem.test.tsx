import React from 'react';
import { shallow } from 'enzyme';
import ContactInfoItem from '../ContactInfoItem';

describe('ContactInfoItem', () => {
  it('Snapshot', () => {
    const component = shallow(
      <ContactInfoItem
        icon="icon"
        iconAlt="icon alternative text"
        content="content"
      />
    );

    expect(component).toMatchSnapshot();
  });
});
