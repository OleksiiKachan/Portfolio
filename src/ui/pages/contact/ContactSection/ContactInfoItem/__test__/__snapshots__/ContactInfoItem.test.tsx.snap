// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactInfoItem Snapshot 1`] = `
<a
  className="contactInfoItem"
>
  <img
    alt="icon alternative text"
    className="contactInfoItem__icon"
    src="icon"
  />
  <p
    className="contactInfoItem__content"
  />
</a>
`;
